src/t_zset.c

```diff
@@ -3067,21 +3067,58 @@ void zcardCommand(client *c) {
     addReplyLongLong(c,zsetLength(zobj));
 }
 
-void zscoreCommand(client *c) {
+/* This generic command implements both ZSCORE and ZMSCORE. */
+void zscoreGenericCommand(client *c, int multi) {
     robj *key = c->argv[1];
     robj *zobj;
     double score;
 
     if ((zobj = lookupKeyReadOrReply(c,key,shared.null[c->resp])) == NULL ||
         checkType(c,zobj,OBJ_ZSET)) return;
 
-    if (zsetScore(zobj,c->argv[2]->ptr,&score) == C_ERR) {
-        addReplyNull(c);
+    /* if more than one member were specified, than return the result in form of a multi-bulk reply */
+    if (!multi) {
+        if (zsetScore(zobj,c->argv[2]->ptr,&score) == C_ERR) {
+            addReplyNull(c);
+        } else {
+            addReplyDouble(c,score);
+        }
+        return;
+    }
+
+    addReplyArrayLen(c,c->argc - 2);
+
+    if (zobj->encoding == OBJ_ENCODING_ZIPLIST) {
+        for (int j = 2; j < c->argc; j++) {
+            if (zzlFind(zobj->ptr,c->argv[j]->ptr,&score) == NULL) {
```

Comment:
@TysonAndre any reason you're not using `zsetScore` like the above?
seems that this is a leftover from the previous PR which was from a time where ZSCORE used to check the encoding itself.

Reply:
It seemed like it may be more efficient to unroll the check of zobj->encoding, since the compiler wouldn't be able to tell zobj->encoding might not change.

This may be a premature optimization since compilers have branch prediction- I didn't look into what redis's C code conventions were for commands that could be used with many variadic args, and can revert this if using the ordinary helpers that hide the encoding is the convention.

Reply:
Updated - I looked at hmgetCommand and readability seems to be prioritized